test/e2e_node/remote/gce/gce_runner.go

```diff
@@ -534,7 +534,10 @@ func (g *GCERunner) createGCEInstance(imageConfig *internalGCEImage) (string, er
 				if err != nil {
 					return "", fmt.Errorf("unable to create temp file %v", err)
 				}
-				defer os.Remove(dataFile.Name()) // clean up
+				defer func() {
+					dataFile.Close()
+					os.Remove(dataFile.Name())
+				}() // clean up
 				if err = os.WriteFile(dataFile.Name(), []byte(item.Value), 0666); err != nil {
 					return "", fmt.Errorf("could not write contents of metadata item into file %v", err)
 				}
```

Comment:
I'd suggest to close it explicitly here. Otherwise it'll remain open (with possibly unwritten data) when gcp will try to read it.

Reply:
It would be even better to close it before calling `os.Writefile` as `os.CreateTemp("", "metadata")` is used here to [get a unique file name](https://man7.org/linux/man-pages/man3/tempnam.3.html).
/hold